pyqasm/visitor.py

```diff
@@ -765,6 +810,67 @@ def _visit_custom_gate_operation(
 
         return result
 
+    def _visit_external_gate_operation(
+        self, operation: qasm3_ast.QuantumGate, inverse: bool = False
+    ) -> list[qasm3_ast.QuantumGate]:
+        """Visit an external gate operation element.
+
+        Args:
+            operation (qasm3_ast.QuantumGate): The external gate operation to visit.
+            inverse (bool): Whether the operation is an inverse operation. Defaults to False.
+
+                            If True, the gate operation is applied in reverse order and the
+                            inverse modifier is appended to each gate call.
+                            See https://openqasm.com/language/gates.html#inverse-modifier
+                            for more clarity.
+
+        Returns:
+            list[qasm3_ast.QuantumGate]: The quantum gate that was collected.
+        """
+
+        logger.debug("Visiting external gate operation '%s'", str(operation))
+        gate_name: str = operation.name.name
+
+        if gate_name in self._custom_gates:
+            # Ignore result, this is just for validation
+            self._visit_custom_gate_operation(operation, inverse=inverse)
+            # Don't need to check if custom gate exists, since we just validated the call
+            gate_qubit_count = len(self._custom_gates[gate_name].qubits)
+        else:
+            # Ignore result, this is just for validation
+            self._visit_basic_gate_operation(operation, inverse=inverse)
+            # Don't need to check if custom gate exists, since we just validated the call
+            _, gate_qubit_count = map_qasm_op_to_callable(operation.name.name)
+
+        op_parameters = [
+            qasm3_ast.FloatLiteral(param) for param in self._get_op_parameters(operation)
+        ]
+
+        self._push_context(Context.GATE)
+
+        modifiers = []
+        if inverse:
+            modifiers = [qasm3_ast.QuantumGateModifier(qasm3_ast.GateModifierName.inv, None)]
+
+        def gate_function(*qubits):
```

Comment:
Why is `op_parameters` not a parameter of the `gate_function`?

Reply:
The way it is currently implemented, `_broadcast_gate_operation()` expects a callable that only takes qubits as input. The intended use is that the non-qubit arguments get evaluated before calling `_broadcast_gate_operation`, e.g. using [`functools.partial`](https://docs.python.org/3/library/functools.html#functools.partial), like [here](https://github.com/qBraid/pyqasm/pull/59/files#diff-5b26d2cb7dbbf506dbaa06405eeb8e9b44a0786e2ac64ede3787b3e96a964207R726). This way, `_broadcast_gate_operation` doesn't need to know about any non-qubit arguments.

But if you prefer the non-qubit arguments to be passed along explictly, I could also easily change that.

Reply:
For now, I have updated the docstring of `_broadcast_gate_operation` to make this point more clear.